src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -249,16 +249,47 @@ async function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourc
   const existingPostRun = moduleConfig.postRun || [];
   (moduleConfig as any).preloadPlugins = [];
 
+  var totalResources = Object.keys(resources.assembly).length + Object.keys(resources.pdb || {}).length + /* dotnet.wasm */ 1;
+
+  if (resourceLoader.bootConfig.resources.runtime.hasOwnProperty('dotnet.timezones.blat')){
+    totalResources++;
+  }
+
+  if (resourceLoader.bootConfig.icuDataMode !== ICUDataMode.Invariant){
+    totalResources++;
+  }
+
+  const targetElement = document.getElementById('app') as HTMLDivElement;
+  const progressElement = document.getElementById('progress') as unknown as SVGCircleElement;
+  const percentageElement = document.getElementById('percentage') as unknown as SVGTextElement;
+  const showProgress = progressElement && percentageElement && targetElement;
+
+  var resourcesLoaded = 0;
+  function setProgress(){
+      resourcesLoaded++;
+      const percentage = resourcesLoaded / totalResources * 100;
+      if (targetElement) {
+        targetElement.style.setProperty('--blazor-load-percentage', percentage + '%');
+      }
+      if (showProgress) {
```

Comment:
Given we're just using this in one place:

```suggestion

  var resourcesLoaded = 0;
  function setProgress(){
      resourcesLoaded++;
      const percentage = resourcesLoaded / totalResources * 100;
      
      if (targetElement) {
        targetElement.style.setProperty('--blazor-load-percentage', percentage + '%');
      }
      
      if (progressElement && percentageElement) {
```